<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>456</width>
    <height>680</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../../icons.qrc">
    <normaloff>:/icons/collection-triangular-iteration.png</normaloff>:/icons/collection-triangular-iteration.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_12">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QPushButton" name="addToCollection_button">
       <property name="statusTip">
        <string>Turn this structure diagram back to structure collections.</string>
       </property>
       <property name="icon">
        <iconset resource="../../../icons.qrc">
         <normaloff>:/icons/collection-structure.png</normaloff>:/icons/collection-structure.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_5">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="load_button">
       <property name="statusTip">
        <string>Triangular iteration data and common structure data collections.</string>
       </property>
       <property name="icon">
        <iconset resource="../../../icons.qrc">
         <normaloff>:/icons/collections.png</normaloff>:/icons/collections.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="clear_button">
       <property name="statusTip">
        <string>Create a new iteration profile.</string>
       </property>
       <property name="icon">
        <iconset resource="../../../icons.qrc">
         <normaloff>:/icons/newfile.png</normaloff>:/icons/newfile.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="save_button">
       <property name="statusTip">
        <string>Save this iteration profile to data collections.</string>
       </property>
       <property name="icon">
        <iconset resource="../../../icons.qrc">
         <normaloff>:/icons/savefile.png</normaloff>:/icons/savefile.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="clipboard_button">
       <property name="statusTip">
        <string>Save this iteration profile to clipboard as a string.</string>
       </property>
       <property name="icon">
        <iconset resource="../../../icons.qrc">
         <normaloff>:/icons/copy.png</normaloff>:/icons/copy.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <widget class="QWidget" name="horizontalLayoutWidget">
      <layout class="QHBoxLayout" name="main_layout">
       <property name="spacing">
        <number>0</number>
       </property>
       <item>
        <widget class="QWidget" name="joint_panel" native="true">
         <property name="maximumSize">
          <size>
           <width>150</width>
           <height>16777215</height>
          </size>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QLabel" name="joint_name_label">
            <property name="text">
             <string>Joint:</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QComboBox" name="joint_name">
              <property name="statusTip">
               <string>Choose a joint tag to see it's status.</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="add_customization">
              <property name="maximumSize">
               <size>
                <width>30</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="statusTip">
               <string>Customize joints and multiple joints option interface.</string>
              </property>
              <property name="icon">
               <iconset resource="../../../icons.qrc">
                <normaloff>:/icons/properties.png</normaloff>:/icons/properties.png</iconset>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="status_label">
            <property name="text">
             <string>Status:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="status">
            <property name="statusTip">
             <string>Each joint should have a solution to find out it's position.</string>
            </property>
            <property name="text">
             <string>N/A</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="solution_label">
            <property name="text">
             <string>Solutions:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="PLAP_solution">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>PLAP</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="PLLP_solution">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>PLLP</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="verticalLayoutWidget_2">
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <widget class="Line" name="line_3">
         <property name="lineWidth">
          <number>3</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSplitter" name="splitter_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <widget class="QWidget" name="layoutWidget">
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="spacing">
            <number>3</number>
           </property>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <item>
              <widget class="QLabel" name="Driver_label">
               <property name="text">
                <string>Drivers:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QListWidget" name="Driver_list">
               <property name="statusTip">
                <string>These joints will setup an revolute input. The number of them as same as DOF.</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_11">
             <item>
              <widget class="QPushButton" name="Driver_add">
               <property name="maximumSize">
                <size>
                 <width>30</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>&lt;&lt;</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="Follower_add">
               <property name="maximumSize">
                <size>
                 <width>30</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>&gt;&gt;</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <widget class="QLabel" name="Follower_label">
               <property name="text">
                <string>Followers:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QListWidget" name="Follower_list">
               <property name="statusTip">
                <string>These joints are on the grounded link, the position of them will generate. So they are don't need to have a solution.</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>&lt;&lt;</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_15">
             <item>
              <widget class="QLabel" name="grounded_label">
               <property name="text">
                <string>Gounded:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QListWidget" name="grounded_list">
               <property name="statusTip">
                <string>Set a link as the ground. Existing solutions will be reset.</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="layoutWidget">
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_7">
               <property name="spacing">
                <number>0</number>
               </property>
               <item>
                <widget class="QLabel" name="Target_label">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="Target_button">
                 <property name="text">
                  <string>Targets</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QListWidget" name="Target_list">
               <property name="statusTip">
                <string>Target points will match as the target path of dimensional synthesis.</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <item>
              <widget class="QPushButton" name="constraints_button">
               <property name="text">
                <string>Constraints</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QListWidget" name="constraint_list">
               <property name="statusTip">
                <string>Four bar loop to apply Gruebler's Equation.</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_10">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_6">
               <item>
                <widget class="QCheckBox" name="show_solutions">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="statusTip">
                  <string>Show triangle sketch on preview window.</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="Expression_list_label">
                 <property name="text">
                  <string>Solutions:</string>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="horizontalSpacer_2">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item>
                <widget class="QPushButton" name="Expression_auto">
                 <property name="statusTip">
                  <string>Auto configure solutions.</string>
                 </property>
                 <property name="icon">
                  <iconset resource="../../../icons.qrc">
                   <normaloff>:/icons/guide.png</normaloff>:/icons/guide.png</iconset>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_8">
               <item>
                <widget class="QListWidget" name="Expression_list">
                 <property name="statusTip">
                  <string>Solutions to find all the positions of joints.</string>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_9">
                 <item>
                  <widget class="QPushButton" name="Expression_clear">
                   <property name="statusTip">
                    <string>Clear all the solutions.</string>
                   </property>
                   <property name="icon">
                    <iconset resource="../../../icons.qrc">
                     <normaloff>:/icons/clean.png</normaloff>:/icons/clean.png</iconset>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <spacer name="verticalSpacer_2">
                   <property name="orientation">
                    <enum>Qt::Vertical</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>20</width>
                     <height>40</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item>
                  <widget class="QPushButton" name="Expression_pop">
                   <property name="statusTip">
                    <string>Remove the last solution.</string>
                   </property>
                   <property name="icon">
                    <iconset resource="../../../icons.qrc">
                     <normaloff>:/icons/arrow_up.png</normaloff>:/icons/arrow_up.png</iconset>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QLabel" name="Link_Expression_label">
             <property name="text">
              <string>Link expression:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="Link_Expression">
             <property name="statusTip">
              <string>Link relative expression.</string>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <widget class="QLabel" name="Expression_label">
             <property name="text">
              <string>Expression:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="Expression">
             <property name="statusTip">
              <string>Triangular iteration expression.</string>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
  <zorder>splitter</zorder>
  <zorder></zorder>
  <zorder>line</zorder>
 </widget>
 <resources>
  <include location="../../../icons.qrc"/>
 </resources>
 <connections/>
</ui>
